lib/AppInfo/Application.php

```diff
@@ -24,30 +25,35 @@
 
 namespace OCA\AnnouncementCenter\AppInfo;
 
-use OCA\AnnouncementCenter\Controller\PageController;
+use OCA\AnnouncementCenter\Dashboard\Widget;
 use OCA\AnnouncementCenter\Manager;
 use OCA\AnnouncementCenter\Model\AnnouncementDoesNotExistException;
 use OCA\AnnouncementCenter\Notification\Notifier;
 use OCP\AppFramework\App;
+use OCP\AppFramework\Bootstrap\IBootContext;
+use OCP\AppFramework\Bootstrap\IBootstrap;
+use OCP\AppFramework\Bootstrap\IRegistrationContext;
 use OCP\Comments\CommentsEntityEvent;
 
-class Application extends App {
+class Application extends App implements IBootstrap {
+	public const APP_ID = 'announcementcenter';
 
 	public function __construct() {
-		parent::__construct('announcementcenter');
-		$container = $this->getContainer();
+		parent::__construct(self::APP_ID);
+	}
 
-		$container->registerAlias('PageController', PageController::class);
+	public function register(IRegistrationContext $context): void {
+		$context->registerDashboardWidget(Widget::class);
 	}
 
-	public function register() {
+	public function boot(IBootContext $context): void {
 		$this->registerNotificationNotifier();
```

Comment:
Could use https://nextcloud-server.netlify.app/classes/ocp-appframework-bootstrap-ibootcontext#method_injectFn for those to use dependency injection for the services inside of the register function

Reply:
Done